code/datums/components/mob_harvest.dm

```diff
@@ -0,0 +1,157 @@
+	SIGNAL_HANDLER
+
+	if(mob_harvest.amount_ready < 1)
+		examine_list += span_notice("[parent] seems like they could use a bit more time.")
+	if(mob_harvest.amount_ready > 1)
+		examine_list += span_notice("[parent] looks like they can be harvested about [mob_harvest.amount_ready] times.")
+	if(mob_harvest.amount_ready == 1)
+		examine_list += span_notice("[parent] looks ready to be harvested.")
+
+///signal called on parent being attacked with an item
+/datum/component/mob_harvest/proc/on_attackby(datum/source, obj/item/used_item, mob/user)
+	SIGNAL_HANDLER
+
+	if(istype(used_item, harvest_tool))
+		mob_harvest.harvest_item(user)
+	if(istype(used_item, fed_item))
+		mob_harvest.remove_wait_time(user)
+		qdel(used_item)
+	return COMPONENT_NO_AFTERATTACK
+
+
+/**
+ * # Harvester item
+ *
+ * Abstract item that is held in nullspace and manages the time it takes to get a new item and creating said item
+ */
+/obj/item/mob_harvest
+	name = "magical mob-powered item generator"
+	desc = "put inside mob to magically make items. you probably shouldn't be able to see this"
+	///typepath of the item you want to harvest
+	var/produced_item_typepath = /obj/item/food/bait
+	///stand-in name for the item
+	var/produced_item_desc = "funny worm"
+	///how much is ready to harvest
+	var/amount_ready = 1
+	///max amount that can be stored between harvests
+	var/max_ready = 1
+	///mob that has the component
+	var/mob/living/harvest_mob
+	///time between item creation
+	var/item_generation_wait = 10 SECONDS
+	///tracked time
+	var/item_generation_time = 0
+	///time to reduce when fed
+	var/item_reduction_time = 2 SECONDS
+
+/obj/item/mob_harvest/Initialize(mapload, harvest_mob)
+	src.harvest_mob = harvest_mob
+	item_generation_time = item_generation_wait
+	START_PROCESSING(SSobj, src)
+	. = ..()
+
+/obj/item/mob_harvest/Destroy()
+	. = ..()
+	STOP_PROCESSING(SSobj, src)
+	harvest_mob = null
+
+/obj/item/mob_harvest/process(delta_time)
+	///only track time if we aren't dead and have room for more items
+	if(harvest_mob.stat != DEAD && amount_ready < max_ready)
+		item_generation_time -= delta_time
+		if(item_generation_time <= 0)
+			item_generation_time = item_generation_wait
+			amount_ready++
+
+/**
+ * Proc called from attacking the component parent with the correct item, reduces wait time between items
+ *
+ * Arguments:
+ * * mob/user - who is trying to do this
+ */
+/obj/item/mob_harvest/proc/remove_wait_time(mob/user)
+	if(amount_ready >= max_ready)
+		to_chat(user, span_warning("[harvest_mob] looks too full to keep feeding!"))
+		return		
+	item_generation_time -= item_reduction_time
+	to_chat(user, span_notice("You feed [harvest_mob]."))
+	return
+
+/**
+ * Proc called from attacking the component parent with the correct item, handles creating the item
+ *
+ * Arguments:
+ * * mob/user - who is trying to do this
+ */
+/obj/item/mob_harvest/proc/harvest_item(mob/user)
+	if(amount_ready >= 1)
```

Comment:
and flip the statement to be `amount_ready < 1` so you can get the one line block out of the way first early returning that